tools/fast/bin/localnet/main.go

```diff
@@ -0,0 +1,350 @@
+	}
+
+	// The genesis process is the filecoin node that loads the miner that is
+	// define with power in the genesis block, and the prefunnded wallet
+	genesis, err := env.NewProcess(ctx, lpfc.PluginName, options, fastenvOpts)
+	if err != nil {
+		exitcode = handleError(err, "failed to create genesis process;")
+		return
+	}
+
+	err = series.SetupGenesisNode(ctx, genesis, genesisMiner.Address, files.NewReaderFile(genesisMiner.Owner))
+	if err != nil {
+		exitcode = handleError(err, "failed series.SetupGenesisNode;")
+		return
+	}
+
+	// Create the processes that we will use to become miners
+	var miners []*fast.Filecoin
+	for i := 0; i < count; i++ {
+		miner, err := env.NewProcess(ctx, lpfc.PluginName, options, fastenvOpts)
+		if err != nil {
+			exitcode = handleError(err, "failed to create miner process;")
+			return
+		}
+
+		miners = append(miners, miner)
+	}
+
+	// We will now go through the process of creating miners
+	// InitAndStart
+	// 1. Initialize node
+	// 2. Start daemon
+	//
+	// Connect
+	// 3. Connect to genesis
+	//
+	// SendFilecoinDefaults
+	// 4. Issue FIL to node
+	//
+	// CreateMinerWithAsk
+	// 5. Create a new miner
+	// 6. Set the miner price, and get ask
+	//
+	// ImportAndStore
+	// 7. Generated some random data and import it to genesis
+	// 8. Genesis propposes a storage deal with miner
+	//
+	// WaitForDealState
+	// 9. Query deal till posted
+
+	var deals []*storagedeal.Response
+
+	for _, miner := range miners {
+		err = series.InitAndStart(ctx, miner)
+		if err != nil {
+			exitcode = handleError(err, "failed series.InitAndStart;")
+			return
+		}
+
+		err = series.Connect(ctx, genesis, miner)
+		if err != nil {
+			exitcode = handleError(err, "failed series.Connect;")
+			return
+		}
+
+		err = series.SendFilecoinDefaults(ctx, genesis, miner, fil)
+		if err != nil {
+			exitcode = handleError(err, "failed series.SendFilecoinDefaults;")
+			return
+		}
+
+		pledge := uint64(10)                    // sectors
+		collateral := big.NewInt(500)           // FIL
+		price := big.NewFloat(0.000000001)      // price per byte/block
+		expiry := big.NewInt(24 * 60 * 60 / 30) // ~24 hours
+
+		ask, err := series.CreateMinerWithAsk(ctx, miner, pledge, collateral, price, expiry)
+		if err != nil {
+			exitcode = handleError(err, "failed series.CreateMinerWithAsk;")
+			return
+		}
+
+		var data bytes.Buffer
+		dataReader := io.LimitReader(rand.Reader, int64(sectorSize))
+		dataReader = io.TeeReader(dataReader, &data)
+		_, deal, err := series.ImportAndStore(ctx, genesis, ask, files.NewReaderFile(dataReader))
+		if err != nil {
+			exitcode = handleError(err, "failed series.ImportAndStore;")
+			return
+		}
+
+		deals = append(deals, deal)
+
+	}
+
+	for _, deal := range deals {
+		err = series.WaitForDealState(ctx, genesis, deal, storagedeal.Posted)
+		if err != nil {
+			exitcode = handleError(err, "failed series.WaitForDealState;")
+			return
+		}
+	}
+
+	if shell {
```

Comment:
this model is a little different than what i expected, which was you run a thing to start the cluster up and then it tells you when it has. then you can run commands against it like shell or similar. i dont feel wed to the other model, just curious about the tradeoffs. my guess is this way we dont have to serialize any state, we can keep it all in process?